### PR TITLE
istio: use labels on namespace instead of iop

### LINKED ISSUE #48: improve istio non flat
enchant istio non flat implement after https://github.com/karmada-io/karmada/issues/2236 released, maybe in v1.3?

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #48

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
istio: use labels on namespace instead of iop
```

### CODE REVIEW

File: pkg/plugin/istio/install.go [hunk 1]
```diff
@@ -100,6 +104,36 @@ func (p *IstioPlugin) ensureNamespaces() error {
...
+	}
+
+	for _, c := range p.allClusters() {
+		network := clusterNetwork{
+			IstioSystemNamespace: istioSystemNamespace,
+			ClusterName:          c,
+			Network:              networkName(c),
+		}
+
+		b, err := templateIstioSystemOverridePolicy(network)
```
[the remark this answers is not in the document]
Comment (zirain) [reply pkg/plugin/istio/install.go:119]: `ClusterOverridePolicy` can be merged to per `network`, per cluster is simple to read

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/plugin/istio/install.go:119]: What do you mean by simple reading? I can see this is read by karmada not by user

[the remark this answers is not in the document]
Comment (zirain) [reply pkg/plugin/istio/install.go:119]: what I'm saying is the logic, it's simple to understand
one `OverridePolicy` for all clusters is impracticable

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**